tests/unit/pipeline/test_ner_re_pipeline.py

```diff
@@ -0,0 +1,390 @@
+    doc = result[0]
+
+    spans = doc["labeled_spans"]
+    assert len(spans) == 3
+
+
+def documents_processor(documents) -> TextBasedDocument:
+    for doc in documents:
+        doc["labeled_spans"].append(LabeledSpan(start=0, end=0, label="empty"))
+    return documents
+
+
+def test_process_pipeline_steps(document):
+    docs = process_pipeline_steps(
+        documents=[document], processors={"add_span": documents_processor}
+    )
+
+    doc = docs[0]
+
+    spans = doc["labeled_spans"]
+    original_spans = document["labeled_spans"]
+    assert len(spans) == 4
+    assert len(original_spans) == 3
+
+    docs = process_pipeline_steps(
+        documents=[document],
+        processors={"add_span": documents_processor},
+        inplace=True,
+    )
+
+    original_spans = document["labeled_spans"]
+    assert len(original_spans) == 4
+
+
+def test_add_annotations_from_other_documents(document, document_with_relations):
+
+    original_relations = document_with_relations["binary_relations"]
+    assert len(original_relations) == 2
+    original_relations_predictions = document_with_relations["binary_relations"].predictions
+    assert len(original_relations_predictions) == 1
+
+    prepared_docs = add_annotations_from_other_documents(
+        docs=[document], other_docs=[document_with_relations], layer_names=["binary_relations"]
+    )
+
+    prepared_doc = prepared_docs[0]
+
+    relations = prepared_doc["binary_relations"]
+    assert len(relations) == 2
+
+    # from predictions
+
+    prepared_docs = add_annotations_from_other_documents(
+        docs=[document],
+        other_docs=[document_with_relations],
+        layer_names=["binary_relations"],
+        from_predictions=True,
+    )
+
+    prepared_doc = prepared_docs[0]
+    relations = prepared_doc["binary_relations"]
+    assert len(relations) == 1
+
+    # to predictions
+
+    prepared_docs = add_annotations_from_other_documents(
+        docs=[document],
+        other_docs=[document_with_relations],
+        layer_names=["binary_relations"],
+        to_predictions=True,
+    )
+
+    prepared_doc = prepared_docs[0]
+    relations = prepared_doc["binary_relations"].predictions
+    assert len(relations) == 3  # adds to previously existing predictions
+
+    # inplace
+
+    prepared_docs = add_annotations_from_other_documents(
+        docs=[document],
+        other_docs=[document_with_relations],
+        layer_names=["binary_relations"],
+        inplace=True,
+    )
+
+    relations = document["binary_relations"]
+    assert len(relations) == 2
+
+
+@dataclasses.dataclass
+class TextDocumentWithEntitiesAndRelations(TextBasedDocument):
+    entities: AnnotationLayer[LabeledSpan] = annotation_field(target="text")
+    relations: AnnotationLayer[BinaryRelation] = annotation_field(target="entities")
+
+
+@pytest.fixture
+def old_document():
+    document = TextDocumentWithEntitiesAndRelations(
+        text="Harry lives in Berlin, Germany. He works at DFKI.", id="tmp"
```

Comment:
Alright, I'll use an example from the tacred dataset

Reply:
I think it would be fine to just "fix" your example: The model should know about the `per:employee_of` relation, but I think it is not detected here because it expects entities that have the labels from the `conll2003` tagset.

Reply:
Hmm I tried that, but it does not seem to work. But I also tried the example input from [here](https://huggingface.co/spaces/pie/Joint-NER-and-Relation-Extraction), but this does not work either. can you investigate, why this works in [the Huggingface Space](https://huggingface.co/spaces/pie/Joint-NER-and-Relation-Extraction), but not in the `NerRePipeline`?

Reply:
All annotations are cleared at the beginning of the pipeline and we did not pass `create_relation_candidates` to re taskmodule, therefore there were no relations or candidates during Classification. After adding the following parameter to `NerRePipeline`, it works.
```python
re_pipeline={"taskmodule_kwargs":{"create_relation_candidates": True}}

```
I'll update the config for ner_re_pipeline accordingly.

Reply:
ah ok, cool, thanks for the catch!